Framework/Kernel/src/Material.cpp

```diff
@@ -297,6 +302,31 @@ double Material::attenuation(const double distance, const double lambda) const {
   return exp(-attenuationCoefficient(lambda) * distance);
 }
 
+/**
+ * @param distance Distance (m) travelled
+ * @param energy KeV to compute the attenuation
+ * @return The dimensionless attenuation factor
+ */
+double Material::xRayAttenuation(const double distance,
+                                 const double energy) const {
+  if (m_xRayAttenuationOverride) {
+    return exp(-m_xRayAttenuationOverride->getAttenuationCoefficient(energy) *
+               distance);
+  } else {
+    throw std::runtime_error("xRayAttenuationProfile override not set");
+    return 1.0;
```

Comment:
I don't think this "return 1.0" line will ever get executed because it follows a throw statement on the line before. Does it compile OK without this line?